modules/programs/afew.nix

```diff
@@ -6,21 +6,134 @@ let
 
   cfg = config.programs.afew;
 
+  dag = config.lib.dag;
+
+  mkIniKeyValue = key: value:
+    let
+      tweakVal = v:
+        if isString v then v
+        else if isList v then concatMapStringsSep ";" tweakVal v
+        else if isBool v then (if v then "true" else "false")
+        else toString v;
+    in
+      "${key}=${tweakVal value}";
+
+  mkDag = filters:
+    let
+      mkNode = nodeName: filter:
+        let
+          before = (filter.before or cfg.defaultBefore);
+          after = (filter.after or cfg.defaultAfter);
+          filterName = (filter.class or "Filter");
+          filterKV = (removeAttrs filter ["after" "before" "class"]) // {
+            "#nodeName" = nodeName;
+            "#before" = before;
+            "#after" = after;
+          };
+        in
+          dag.entryBetween before after (nameValuePair filterName filterKV);
+    in
+      mapAttrs mkNode filters;
+
+  /* Return a topologically sorted list of DAG entries encapsulating
+     name value pairs.
+  */
+  reorderFilters = filters:
+    (dag.topoSort (mkDag filters)).result;
```

Comment:
If you leave this as
```
  reorderFilters = filters: dag.topoSort (mkDag filters);
```
then you could have an assertion
```
assertions = [
  {
    assertion = reorderFilters ? result;
    message = "The 'programs.afew.filters' option contains a circular dependency.";
  }
];
```

Reply:
Same as above with `warnings`, I'm not sure how to use `assertions`.